### PR TITLE
Members page contributors/watchers (UI only)

### LINKED ISSUE #3379: Split members on the members page to contributors & watchers - UI only
Root view: Contributors (including those with permissions) on the top, and watchers below.

Domain (sub-domains) view (no watchers, just contributors and admins).

Pagination should be included as per the specs.

https://www.figma.com/file/qQoBbyyDXkuHwoqHIOGjEu/Colony-Simplified?node-id=6736%3A92817

Screenshot 2022-05-19 at 12 15 56

### PR DESCRIPTION
**Pull request by dit7ya**
## Description

This PR introduces a MembersSection component (in the Members page) that can be reused for contributors/watchers etc. It also adds the UI and styling for the said component.

The base branch for this PR is set as the logic branch for this feature. I was not sure if this approach or asking people to review changes from just my commit is more canonical.

*Request for comments*

- What do think about the level of decoupling of the MembersSection component? I am not super happy about it but since many related features are yet to be merged/implemented there was no easy way to decide the appropriate code organization.

*Things not implemented* (as those are part of other issues despite being in the same spec)

- The columns redesign of the Members
- The Title Bar 
- Search bar 
- Right aside things

Screenshot 2022-05-20 at 16-29-02 Members Colony - Pequod

Resolves #3379.

### CODE REVIEW

**Comment by danbr** [comment]
@dit7ya I'm a bit confused, you've listed what you haven'd done, but to help me review the PR i'm more interested in what you have implement or what you intended to include in the PR. Then i know where to focus my attention.

I dont know if its supposed to be included here, but the list items in the `MembersSection` are not at all like the figma specs.

Figma:
Screenshot 2022-05-18 at 20 35 11

Dapp:
Screenshot 2022-05-18 at 20 34 51

Also - I dont think any logic is included in this PR, but how about the UI for the columns, is that supposed to be implemented or in a different PR?
Screenshot 2022-05-18 at 20 37 22

File: src/modules/dashboard/components/Members/MembersSection.tsx
Comment (rdig) [inline src/modules/dashboard/components/Members/MembersSection.tsx:57]: So I'm not really sure but I think pagination specs look different for this page now.

Please check with Product to figure out what these are.

File: src/modules/dashboard/components/Members/MembersSection.tsx
Comment (rdig) [inline src/modules/dashboard/components/Members/MembersSection.tsx:87]: So the members list needs to be refactored to account for the new spec changes.

You cannot simply reuse it in this instance.

**Review by rdig — CHANGES_REQUESTED** [review]
Code wise this is pretty ok, however it's missing the `MembersList` refactor, as others have already pointed out.

**Comment by chinins** [comment]
@dit7ya, please rebase and update this branch based on my `feature/3377-add-contributors-watchers-data-query`. I'm converting it to draft until then as it's not ready yet.

**Comment by dit7ya** [comment]
@chinins Going to rebase it soon. @rdig @danbr The members column redesign is being worked on in #3392 and is not in the scope of the linked issue, I have confirmed with Olga.

**Comment by dit7ya** [comment]
Rebased it @olga, opening it for review.

Reviewers: Please take a look at the RFC in the description of this PR.

File: src/modules/dashboard/components/Members/MembersSection.tsx
Comment (chinins) [inline src/modules/dashboard/components/Members/MembersSection.tsx:66]: This whole block is just repeating the code. I would just lift the `extraItemContent` prop to the `Members` component. Then you don't need repeating or `contributor` check.

**Review by chinins — CHANGES_REQUESTED** [review]
Please, lift up `extraItemContent` prop to the level above - as per my comment.

File: src/modules/dashboard/components/Members/MembersSection.tsx
Comment (dit7ya) [reply src/modules/dashboard/components/Members/MembersSection.tsx]: Done, thanks! I was confused about the coupling but I guess this makes more sense.

File: src/modules/dashboard/components/Members/Members.tsx
Comment (chinins) [inline src/modules/dashboard/components/Members/Members.tsx:52]: I would prefer if you moved the title and description into the `MembersSection` component and pass less props. `Members` component is already huge and the goal is to move as much as possible from it.

**Review by chinins — CHANGES_REQUESTED** [review]
Looking better, just a couple of more changes.

File: src/modules/dashboard/components/Members/MembersSection.tsx
Comment (chinins) [inline src/modules/dashboard/components/Members/MembersSection.tsx:60]: You are not actually using this data or query so it should be completely removed. I would have the `isLoadingData` prop on `LoadMoreButton` either as optional or always false in our case.

**Review by chinins — APPROVED** [review]
There are a couple of comments but preapproving.

**Comment by chinins** [comment]
Please, don't merge it into my branch `[feature/3377]` branch, just wait until I merge my branch into the feature branch.

**Pull request merged**
